src/Components/Blazor/Blazor/test/Hosting/RootComponentMappingTest.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+
+using System;
+using System.Text;
+using Microsoft.AspNetCore.Components.Routing;
+using Microsoft.AspNetCore.Testing;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Blazor.Hosting
+{
+    public class RootComponentMappingTest
+    {
+        [Fact]
+        public void Constructor_ValidatesComponentType_Success()
+        {
+            // Arrange
+            // Act
+            var mapping = new RootComponentMapping(typeof(Router), "test");
+
+            // Assert (does not throw)
+            GC.KeepAlive(mapping);
```

Comment:
You could do `Assert.Null(Record.Exception(() => new RootComponentMapping(typeof(Router), "test"));`. It was surprising to see a GC.KeepAlive call here when no linking is involved.

Reply:
eh, that's pretty complicated.